src/GCodes/GCodes2.cpp

```diff
@@ -2079,14 +2079,57 @@ bool GCodes::HandleMcode(GCodeBuffer& gb, const StringRef& reply) THROWS(GCodeEx
 					}
 #endif
 
-					if (result != GCodeResult::error)
+#if SUPPORT_MQTT
+					if ((type & MqttMessage) && (result != GCodeResult::error))
 					{
-						// Append newline and send the message to the destinations,
-						// except for MqttMessage
-						if (type != MqttMessage)
+						String<MaxGCodeLength> topic;
+						gb.MustSee('T');
+						gb.GetQuotedString(topic.GetRef());
+
+						int32_t qos = 0;
+						if (gb.Seen('Q'))
+						{
+							qos = gb.GetIValue();
```

Comment:
Please use GetLimitedUIValue to get a value between 0 and 2, then we can eliminate the test of the value of qos and the error message. We have very little free flash memory left on Duet 2 so we need to avoid using space for dedicated range tests wherever possible.

Reply:
Fixed. Didn't know about GetLimitedUIValue, thanks for the heads up! Actually used TryGetLimitedUIValue for the optional parameters.